ras_party/controllers/queries.py

```diff
@@ -199,10 +201,19 @@ def search_businesses(search_query, session):
     filters.append(and_(*trading_as_filters))
 
     bound_logger.unbind('search_query')
-    return session.query(BusinessAttributes.name, BusinessAttributes.trading_as, Business.business_ref)\
+    query = session.query(BusinessAttributes.name, BusinessAttributes.trading_as, Business.business_ref)\
         .join(Business)\
         .filter(and_(or_(*filters), BusinessAttributes.collection_exercise.isnot(None)))\
-        .distinct().all()
+        .distinct().order_by(BusinessAttributes.name)          # Build the query
+
+    results = query.limit(limit).offset((page-1)*limit).all()  # Execute the query
+    if page == 1 and len(results) < limit:
+        total_business_count = len(results)
+    else:
+        count_q = query.statement.with_only_columns([func.count(distinct(Business.business_ref))]).order_by(None)
```

Comment:
nice job with the only columns, I imagine it makes the search more efficient.
Do you need the order_by(None)?

Reply:
I think it could left  it off but I think its an explicit instruction to the DBM  to ignore the sort column and hence gives it the option of selecting the most efficient way of getting the result without worrying about sort order .